cryptographic_estimators/MQEstimator/MQAlgorithms/f5.py

```diff
@@ -76,41 +76,31 @@ def degree_of_polynomials(self):
         """
         return self._degrees
 
-    def time_complexity(self):
+    def _compute_time_complexity(self, parameters: dict):
         """
-        Return the time complexity of the F5 algorithm
+        Return the time complexity of the algorithm for a given set of parameters
 
         EXAMPLES::
 
             sage: from cryptographic_estimators.MQEstimator.MQAlgorithms.f5 import F5
             sage: from cryptographic_estimators.MQEstimator.mq_problem import MQProblem
             sage: E = F5(MQProblem(n=10, m=15, q=3))
             sage: E.time_complexity()
-            19.934452517671986
+            29.077131123796804
 
         TESTS::
 
             sage: F5(MQProblem(n=10, m=12, q=5)).time_complexity()
-            25.934452517671986
+            35.53502411058788
 
         """
-        memory = self.memory_complexity()
-        if self.bit_complexities:
-            memory = self.problem.to_bitcomplexity_memory(memory)
-        if memory > self.problem.memory_bound:
-            self._time_complexity = inf
-            self._memory_complexity = inf
-            return inf
-
-        if self._time_complexity is None:
-            if self.problem.is_overdefined_system():
-                self._time_complexity = self._time_complexity_semi_regular_system()
-            else:
-                self._time_complexity = self._time_complexity_regular_system()
+        if self.problem.is_overdefined_system():
+            self._time_complexity = self._time_complexity_semi_regular_system()
+        else:
+            self._time_complexity = self._time_complexity_regular_system()
 
-            if self.complexity_type == ComplexityType.ESTIMATE.value:
-                self._time_complexity = max(
-                    self._time_complexity, self._time_complexity_fglm())
+        self._time_complexity = max(
+            self._time_complexity, self._time_complexity_fglm())
 
         return self._time_complexity
```

Comment:
this function should compute the time complexity and return it. NEVER set _time_complexity.

Reply:
this is all done by the base class time_complexity function